agent/s3.go

```diff
@@ -39,17 +41,18 @@ func (e *credentialsProvider) Retrieve() (creds credentials.Value, err error) {
 	creds.SessionToken = os.Getenv("BUILDKITE_S3_SESSION_TOKEN")
 
 	if creds.AccessKeyID == "" {
-		err = errors.New("BUILDKITE_S3_ACCESS_KEY_ID or BUILDKITE_S3_ACCESS_KEY not found in environment")
+		return credentials.Value{}, errors.New("BUILDKITE_S3_ACCESS_KEY_ID or BUILDKITE_S3_ACCESS_KEY not found in environment")
 	}
+
 	if creds.SecretAccessKey == "" {
-		err = errors.New("BUILDKITE_S3_SECRET_ACCESS_KEY or BUILDKITE_S3_SECRET_KEY not found in environment")
+		return credentials.Value{}, errors.New("BUILDKITE_S3_SECRET_ACCESS_KEY or BUILDKITE_S3_SECRET_KEY not found in environment")
 	}
 
 	e.retrieved = true
-	return
```

Comment:
i think named returns are my least favourite feature in all of go. i have removed them from the files i've touched in this PR.